lib/SyTest/Federation/Server.pm

```diff
@@ -96,6 +103,34 @@ sub get_event
    return $event;
 }
 
+sub get_auth_chain
+{
+   my $self = shift;
+   my @event_ids = @_;
+
+   my %events_by_id = map { $_ => $self->get_event( $_ ) } @event_ids;
+
+   my @auth_chain = @event_ids;
+
+   while( @event_ids ) {
+      my $event = $events_by_id{shift @event_ids};
+
+      my @auth_ids = map { $_->[0] } @{ $event->{auth_events} };
```

Comment:
It would be nice if auth_events were a struct with named values, rather than knowing by convention that the 0th element happens to be an array of auth event IDs.